broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java

```diff
@@ -84,7 +84,7 @@ public RemotingCommand processRequest(ChannelHandlerContext ctx,
 
     @Override
     public void asyncProcessRequest(ChannelHandlerContext ctx, RemotingCommand request, RemotingResponseCallback responseCallback) throws Exception {
-        asyncProcessRequest(ctx, request).thenAcceptAsync(responseCallback::callback, this.brokerController.getSendMessageExecutor());
+        asyncProcessRequest(ctx, request).thenAcceptAsync(responseCallback::callback, this.brokerController.getPutMessageFutureExecutor());
```

Comment:
The callback is an async netty operation (ctx.writeAndFlush). I suggest use thenAccept() method, the writeAndFlush may run in these threads:

1. sendMessageExecutor
2. the flush thread
3. the transfer thread

Or use a single thread executor.

Reply:
The original method was thenAccept, but it was changed to thenaAcceptAsync because of the issue #1879 #1881